Documentation building is broken due to numpy-2 and tensorflow

Documentation building is broken, for both main and PR branches.

Latest version appearing in online docs is 0.4.0 instead of 0.4.3.

See an example of failing action: https://github.com/huggingface/evaluate/actions/runs/10826041642/job/30036188975
```
TypeError: Unable to convert function return value to a Python type! The signature was
	() -> handle
Error: Process completed with exit code 1.
```

--- a/setup.py
+++ b/setup.py
@@ -79,6 +79,7 @@ TESTS_REQUIRE = [
     "pytest-datadir",
     "pytest-xdist",
     # optional dependencies
+    "numpy<2.0.0",  # tensorflow requires numpy < 2
     "tensorflow>=2.3,!=2.6.0,!=2.6.1, <=2.10",
     "torch",
     # metrics dependencies

